src/components/map/ha-entity-marker.ts

```diff
@@ -26,7 +29,9 @@ class HaEntityMarker extends LitElement {
                 "background-image": `url(${this.entityPicture})`,
               })}
             ></div>`
-          : this.entityName}
+          : this.entityIcon
+            ? html`<ha-icon .icon=${this.entityIcon}></ha-icon>`
```

Comment:
Cant we just use `ha-state-icon`? That will handle it all.

Reply:
I kinda tried this once but didn't get it to work. 

`ha-state-icon` needs `hass`, and I think the way the marker is constructed in ha-map and the html template is passed to `map.addLayer()` makes that not work for some reason (currently everything to ha-entity-marker is passed as an attribute instead of a property?), when I tried to add a hass property to the marker it always ended up being undefined when that item rendered. 

Probably need some advanced JS I don't understand to make that work correctly?

Reply:
This should kinda work, it will not update `hass` when it changes though, but that could also be added if needed:

```ts
      const entityMarker = document.createElement("ha-entity-marker");
      entityMarker.hass = this.hass;
      entityMarker.entityId = getEntityId(entity);
      entityMarker.entityName = entityName;
      entityMarker.entityPicture = entityPicture
        ? this.hass.hassUrl(entityPicture)
        : "";
      if (typeof entity !== "string") {
        entityMarker.entityColor = entity.color;
      }

      // create marker with the icon
      const marker = Leaflet.marker([latitude, longitude], {
        icon: Leaflet.divIcon({
          html: entityMarker,
          iconSize: [48, 48],
          className: "",
        }),
        title: title,
      });
```

Another option would be to extend `Leaflet.divIcon` with `lit-html` support, so we can write `lit-html` templates in the `html` option.

Reply:
Thanks, this suggestion seems to work well. 

Now have working icons for all entities 👍 (even updates correctly when the state changes)